apps-rendering/src/client/article.ts

```diff
@@ -87,34 +105,116 @@ function followToggle(topic: Topic): void {
 	});
 }
 
-function topicClick(e: Event): void {
-	const follow = document.querySelector('.js-follow');
+function notificationsFollowClick(e: Event): void {
+	const follow = document.querySelector('.js-follow-notifications');
 	const topic = getTopic(follow);
+	if (topic) {
+		followToggle(
+			topic,
+			'.js-follow-notifications-status',
+			FollowNotificationStatus,
+			notificationsClient,
+		);
+	}
+}
 
+function tagFollowClick(e: Event): void {
+	const follow = document.querySelector('.js-follow-tag');
+	const topic = getTopic(follow);
 	if (topic) {
-		followToggle(topic);
+		followToggle(
+			topic,
+			'.js-follow-tag-status',
+			FollowTagStatus,
+			tagClient,
+		);
 	}
 }
 
+function conditionallyRenderFollowTagComponent(
+	topic: Topic,
+	followTagStatus: Element | null,
+	followTag: Element | null,
+): void {
+	const checkBridgetCompatibilty = (
+		bridgetVersion: Optional<string>,
+	): boolean =>
+		bridgetVersion
+			.map((versionString) => compare(versionString, '2.5.0', '>='))
+			.withDefault(false);
+
+	const isBridgetCompatible = getBridgetVersion().then((version) => {
+		return checkBridgetCompatibilty(version);
+	});
+
+	const isMyGuardianEnabled = environmentClient.isMyGuardianEnabled();
+
+	const tagIsFollowingState = tagClient.isFollowing(topic);
+
+	void Promise.all([
+		isBridgetCompatible,
+		isMyGuardianEnabled,
+		tagIsFollowingState,
+	])
+		.then(
+			([
+				isBridgetCompatible,
+				isMyGuardianEnabled,
+				tagIsFollowingState,
+			]) => {
+				isBridgetCompatible &&
+					isMyGuardianEnabled &&
+					followTagStatus &&
+					ReactDOM.render(
+						h(FollowTagStatus, {
+							isFollowing: tagIsFollowingState,
+							contributorName: topic.displayName,
+						}),
+						followTagStatus,
+					);
+
+				followTag?.addEventListener('click', tagFollowClick);
```

Comment:
unsure which you prefer, but it makes sense to me to keep the name's the same e.g `followTag` & `followTagClick`

or

`tagFollow` and `tagFollowClick`